src/state.ts

```diff
@@ -0,0 +1,13 @@
+import { Store as VuexStore } from "vuex";
+
+export type TState<T> = T extends (...arg: any[]) => infer R ? R : T;
+
+export function wrapState<TRootState>(
```

Comment:
I like the idea, but this wrapper doesn't provide any type-safety. It's returning an implicit `any` value, which won't cause compilation errors but also won't protect any downstream code.

Reply:
On closer review, it looks like this is a non-issue so long as this function stays internal.